core/trino-main/src/test/java/io/trino/operator/TestHashAggregationOperator.java

```diff
@@ -682,15 +681,10 @@ public void testSpillerFailure()
                 blockTypeOperators,
                 false);
 
-        try {
-            toPages(operatorFactory, driverContext, input);
-            fail("An exception was expected");
-        }
-        catch (RuntimeException expected) {
-            if (!nullToEmpty(expected.getMessage()).matches(".* Failed to spill")) {
-                fail("Exception other than expected was thrown", expected);
-            }
-        }
+        assertThatThrownBy(() -> toPages(operatorFactory, driverContext, input))
+                .isInstanceOf(RuntimeException.class)
+                .hasCauseInstanceOf(IOException.class)
+                .hasMessageEndingWith("Failed to spill");
     }
```

Comment:
hasCauseInstanceOf(IOException.class) was added, it is not fully clear to me if this is the only possible cause, could you please comment?